ios/RNSScreen.mm

```diff
@@ -2,57 +2,83 @@
+#import <react/renderer/components/rnscreens/RCTComponentViewHelpers.h>
+#import <rnscreens/RNSScreenComponentDescriptor.h>
+#import "RCTFabricComponentsPlugins.h"
+#import "RNSConvert.h"
+#import "RNSScreenStackHeaderConfigComponentView.h"
+#else
 #import <React/RCTTouchHandler.h>
+#import "RNSScreenStack.h"
+#endif
+
+#import <React/RCTShadowView.h>
 #import <React/RCTUIManager.h>
+#import "RNSScreenStackHeaderConfig.h"
 
-@interface RNSScreenView () <UIAdaptivePresentationControllerDelegate, RCTInvalidating>
+@interface RNSScreenView ()
+#ifdef RN_FABRIC_ENABLED
+    <RCTRNSScreenViewProtocol, UIAdaptivePresentationControllerDelegate>
+#else
+    <UIAdaptivePresentationControllerDelegate, RCTInvalidating>
+#endif
 @end
 
 @implementation RNSScreenView {
   __weak RCTBridge *_bridge;
-  RNSScreen *_controller;
+#ifdef RN_FABRIC_ENABLED
+  RCTSurfaceTouchHandler *_touchHandler;
+  facebook::react::RNSScreenShadowNode::ConcreteState::Shared _state;
+#else
   RCTTouchHandler *_touchHandler;
   CGRect _reactFrame;
+#endif
+}
+
+#ifdef RN_FABRIC_ENABLED
+- (instancetype)initWithFrame:(CGRect)frame
+{
+  if (self = [super initWithFrame:frame]) {
+    static const auto defaultProps = std::make_shared<const facebook::react::RNSScreenProps>();
+    _props = defaultProps;
+    [self initCommonProps];
+  }
+
+  return self;
 }
+#endif
 
 - (instancetype)initWithBridge:(RCTBridge *)bridge
 {
   if (self = [super init]) {
     _bridge = bridge;
-    _controller = [[RNSScreen alloc] initWithView:self];
-    _stackPresentation = RNSScreenStackPresentationPush;
-    _stackAnimation = RNSScreenStackAnimationDefault;
-    _gestureEnabled = YES;
-    _replaceAnimation = RNSScreenReplaceAnimationPop;
-    _dismissed = NO;
-    _hasStatusBarStyleSet = NO;
-    _hasStatusBarAnimationSet = NO;
-    _hasStatusBarHiddenSet = NO;
-    _hasOrientationSet = NO;
-    _hasHomeIndicatorHiddenSet = NO;
+    [self initCommonProps];
   }
 
   return self;
 }
 
-- (void)reactSetFrame:(CGRect)frame
+- (void)initCommonProps
 {
-  _reactFrame = frame;
-  UIViewController *parentVC = self.reactViewController.parentViewController;
-  if (parentVC != nil && ![parentVC isKindOfClass:[RNScreensNavigationController class]]) {
-    [super reactSetFrame:frame];
-  }
-  // when screen is mounted under RNScreensNavigationController it's size is controller
-  // by the navigation controller itself. That is, it is set to fill space of
-  // the controller. In that case we ignore react layout system from managing
-  // the screen dimensions and we wait for the screen VC to update and then we
-  // pass the dimensions to ui view manager to take into account when laying out
-  // subviews
+  _controller = [[RNSScreen alloc] initWithView:self];
+  _stackPresentation = RNSScreenStackPresentationPush;
+  _stackAnimation = RNSScreenStackAnimationDefault;
+  _gestureEnabled = YES;
+  _replaceAnimation = RNSScreenReplaceAnimationPop;
+  _dismissed = NO;
```

Comment:
All those props below can probably be not set since they should be `NO` by default. Could you check it and remove those assignments if so?

Reply:
I see your point: these are most likely redundant lines of code and I agree. However I see no harm in making this initialisation explicit, it only improves the readability. There should be no impact on performance / robustness. What do you think?

Reply:
As for readability, I have no strong opinion. We can keep them like this.